### PR TITLE
fix filenotfound errors on pathlib loading

### PR DESCRIPTION
**Pull request by ankith26**
Fixes bug where `FileNotFoundError`s are not displayed right when file loading fails from pathlib objects. Also does a bit of code cleanup in the function handling the errors

### CODE REVIEW

File: src_c/rwobject.c [hunk 1]
```diff
@@ -700,90 +662,83 @@ _pg_rw_read(SDL_RWops *context, void *ptr, size_t size, size_t maxnum)
...
+                        " '%S'.", obj, cwd);                       
+
+    Py_DECREF(cwd);
+    Py_DECREF(isabs);
+    return NULL;
+
+error:
+    Py_XDECREF(cwd);
+    Py_XDECREF(isabs);
+    PyErr_Format(PyExc_FileNotFoundError, "No such file or directory: '%S'.", obj);
```
Comment (Starbuck5) [inline src_c/rwobject.c:741]: It's a bit strange to have a state labeled error that is fine to go to in normal execution.

Maybe it should be named like `simple_case` or something?

**Comment by Starbuck5** [comment]
This is a lot of code change to fix the issue, but I see it also simplifies some code and removes from python <3.6 compatibility, which is cool.

Just in case, for the future, I would also like a test in the pathlib loading test that asserts it raises a FileNotFound error when the file is nonexistent.

File: src_c/rwobject.c (hunk 1 above)
Comment (ankith26) [reply src_c/rwobject.c:741]: yeah, true

**Comment by ankith26** [comment]
The old code has unnecessary nesting which was painful on readability and that's why it was harder to catch the bug. Once I unnested and simplified the code it was an easy fix

> Also, could you please highlight to me what of this exactly fixes the problem?

The issue was that the code was doing `PyUnicode_Check(obj)` before, so when `obj` is pathlib object or bytes object, instead of erroring cleanly, it goes into the file object handling side, which then later fails with `TypeError: Not a file object`

The check is not required at all because we are sure that `obj` should be `str | bytes | os.PathLike[str] | os.PathLike[bytes]` at that stage

**Review by MyreMylar — APPROVED** [review]
Tested locally and can confirm it fixes the error with Path() objects error messages.

👍

**Review by illume — APPROVED** [review]
👍 thanks

**Pull request merged**